Streamlining definition and reading of tables

pbuehler opened the pull request:
. default tree names are deduced from the table origin and description

origin O, description T:S
O==AOD
treename = O2t
O == RN2
treename = Run2t

There is currently 1 exception to this. Since the description can have maximum 16 characters (including the ending \0, thus effectively only 15 characters) the description MCCOLLISIONLABEL does not match. Instead MCCOLLISLABEL is used, although the corresponding tree name is O2mccollisionlabel.

bovulpes committed:
MID: add library MIDWorkflow

dstocco committed:
Fix clang-format in MID Encoder

mpuccio committed:
[ITSCA] Tracker performance tuning

Finer grid size gives ~1.4x speedup in the tracker. However, the physics performance is quite stable with efficiency oscillations between two values that differ of ~0.2 per mille.
This are currently being investigated.

mpuccio committed:
Fix CUDA code compilation

preghenella committed:
External trigger configuration moved to configurable params

preghenella committed:
External generator configuration moved to configurable params

preghenella committed:
Workflow for plug-in TOF analyses of compressed data

sawenzel committed:
More efficient structures for MCTruth sharing and ROOT IO

This commit achieves/provides the following:

* A new class ConstMCTruthContainer, allowing to
  share **read-only** MC labels in a flat buffer and
  in shared memory (using the DPL->make mechanism).
  This avoids going through ROOT IO when communicating between
  DPL devices and reduces the overall memory usage and CPU time.

* A new class IOMCTruthContainerView, allowing to overcome
  the size limitation of labels when writing to a TTree entry.
  This is achieved by splitting the buffer into several members, which
  ROOT serializes separately.

* A test workflow is provided, exchanging MClabels between a couple
  of DPL processes. The workflow can run with the existing as well
  the new way of sharing labels.

pbuehler committed:
Streamlining definition and reading of tables

pbuehler committed:
Streamlining definition and reading of tables

mfasDa committed:
[EMCAL-671] Fix leftover geometry path which is missing the barrel mothervolume

mfasDa committed:
[EMCAL-671] Rename "barrel" to "barrel_1"

shahor02 committed:
Init mag.field before initializing generator

aalkin commented on Framework/Core/src/AnalysisDataModel.cxx:
```diff
@@ -0,0 +1,111 @@
+#include "Framework/Logger.h"
+#include <boost/algorithm/string.hpp>
+
+namespace o2
```
You can inline this, `namespace o2::aod::datamodel {}`

pbuehler replied:
done

aalkin commented on Framework/Core/src/AnalysisDataModel.cxx:
```diff
@@ -0,0 +1,111 @@
+  // AOD / RN2
+  if (origin == "AOD") {
+    if (description == "TRACK:PAR") {
+      return columnNamesTrait(decltype(StoredTracksMetadata{})::table_t::persistent_columns_t{});
```
Why you need decltype here? It should be enough to have `typename StoredTracksMetadata::table_t::persistent_columns_t{}`

pbuehler replied:
does also work - done.

ktf committed:
DPL: do proper endOfStream for test_RegionInfoCallbackService (#4296)

Somehow breaks only during the threading setup

shahor02 committed:
TrackPar(Cov) can be invalidated by calling invalidate() method

This will set its mX to meaningless value, which can be checked by isValid() method.

shahor02 committed:
Fast DCAFitterN method to get track params at PCA w/o full propagation

Only a TrackPar copy will be propagated and returned, in case of propagation failure TrackPar::isValid() will return false.

aalkin committed:
DPL Analysis: reintroduce TableTransform base class (#4295)

dsekihat committed:
add dilepton ee task (#4276)

pbuehler committed:
Simplifications

pbuehler committed:
Simplifications

lbariogl committed:
Add task for nuclei (#4301)

pbuehler committed:
clang-format

pbuehler committed:
clang-format

shahor02 committed:
Dedicate field creator method for construction from nominal rounded L3 field value

aalkin committed:
DPL Analysis: drop arrow v0.17 compatibility (#4304)

ktf committed:
DPL: add RawDeviceService::waitFor method (#4303)

This avoids exposing FairMQDevice.h to the user in a few more places.
Every inclusion of FairMQDevice.h results in 1GB of RAM and 2 seconds
of compilations in my naive tests.

shahor02 committed:
Add method to enforce oldest slot finalization/removal

ktf committed:
DPL Foundation: use intrinsict for pack element, when available (#4312)

shahor02 committed:
Base class for compact index with provenance and flags info

shahor02 committed:
Global track index for vertex->track reference

dsekihat committed:
update eID (#4314)

lbariogl committed:
Add filters to nuclei task (#4311)

aalkin committed:
DPL Analysis: adapt Attach/Extend to new Join (#4307)

ktf committed:
DPL Analysis: move ROOT helpers in a separate file

ktf committed:
Add script to merge multiple -ftime-trace outputs.

ktf committed:
Ignore .clangd folder

Cache for Language Server compilations

njacazio committed:
Use CCDB headers to get SOR timestamp for run (#4305)

- Update timestamp table maker task
- Caching requested timestamps for later use
- Remove old utilities for Run to Timestamp conversion

shahor02 committed:
Fix: protection for processing empty sector

ekryshen committed:
Event selection adapted to MC (#4319)

ktf committed:
DPL Foundation: speedup has_type_at (#4315)

Use multiple inheritance rather than recursion to speedup compilation and
hopefully reduce memory usage.

ktf committed:
DPL: hide a few more headers (#4310)

dstocco committed:
Use the new raw data decoders in the checks (#4321)

Use the new raw data decoders in the checks

- Move checkers to a dedicated directory
- Use a lighter raw file reader
- Improve the raw dumper, allowing to dump the decoded data on demand

mario-krueger committed:
add some histogram helpers (#4239)

aalkin committed:
DPL Analysis: first draft for more generic index binding (#4325)

shahor02 committed:
Make TPC fwd/bwd time margins float

shahor02 committed:
Enforce minimal fwd/bwd delta for CE crossing TPC tracks

shahor02 committed:
Add BunchFilling methods to get 1st and last filled bunch

shahor02 committed:
Add InteractionRecord method to get diff. in ns to other IR

shahor02 committed:
PrimaryVertex min/max IR coverage support

shahor02 committed:
Validate P.Vertex by bunch filling, set min/max IR

shahor02 committed:
P.Vertex-tracks matching class + related workflows update

At the moment the pr.vertex - track matching is done in time only (except vertex fit contributor TPC/ITS
tracks), for every vertex list of global indices (VtxTrackIndex) is added, providing the source (global,
or ITS or TPC track), ID in the source and bit flags for contributors and ambigous assignment.

shahor02 committed:
Reject TPCITS match if corr. time is outside of ITS ROF

knopers8 committed:
[QC-357] Move Data Sampling from Framework to Utilities (#4135)

nzardosh committed:
adding D0 selector (#4294)

shahor02 committed:
Validate ITS/TPC match by Bunch filling scheme (except cosmics mode)

mfasDa committed:
[EMCAL-668] Add track position on EMCAL surface to TRACKEXTRA table (#4306)

sawenzel committed:
TRD digitization: Application of new labels container structure

ktf committed:
DPL: remove spurious message from Monitor::Cleanup (#4334)

Requires FairMQ 1.4.23

victor-gonzalez committed:
Incorporated full pair processing (#4329)

* Incorporated full pair processing
- basis for configurable enhanced cuts support
- parallel execution per centrality/multiplicity range
- filtered per centrality/multiplicity range
- task only receive collisions within centrality/multiplicity range
- single configurable binning structure
- fully implemented at task level
- single configurable event selection structure

mario-krueger committed:
HistHelpers: improved histogram collection and hist builder (#4330)

- support TProfiles
- support TList based collections
- support weighted fills
- allow in-place allocation of root histograms with new
- rename HistBuilder -> Hist and remove root pointer member
- allow copying Hist objects and adding axes from existing Hist

jgrosseo committed:
some refactoring in correlation container (#4328)

dstocco committed:
Improve performance of MID raw data decoding without the user logic (#4341)

jgrosseo commented on Framework/Core/src/AnalysisDataModel.cxx:
```diff
@@ -0,0 +1,105 @@
+    } else if (description == "TRACK:PARCOV") {
+      return columnNamesTrait(typename StoredTracksCovMetadata::table_t::persistent_columns_t{});
+    } else if (description == "TRACK:EXTRA") {
+      return columnNamesTrait(typename TracksExtraMetadata::table_t::persistent_columns_t{});
```
After a discussion with Paul, my understanding is that these lines are only needed for tables where we read the table partially (at the moment the Track table). The other lines are not needed, but don't harm either. I am worried that they may give the impression that new tables will not be supported.
@ktf @aalkin What is your opinion?

aalkin replied:
This is an unfortunate consequence of matching runtime information to static types, one way or another you need to hardcode this correspondence between the two. Additionally, this may allow us to have versioning for trees/tables that we are reading, which requires to have a list of branches/columns known, similar to partially read trees.

jgrosseo replied:
My understanding is that we only need to hardcode it for Tracks.
My question is if we shouldn't drop it for all other tables, where we do *not* need to hardcode it. At least the code also works for user-provided tables without changing the AOD reader.

ktf replied:
Indeed. For whatever does not have `:` in the name, we can simply assume that the columns are the ones which are found in the tree, no?

ktf replied:
I was also thinking that in the long term we can remove it also for the partial tables (e.g. TRACKs), by having a static linked list (a C one, not an std::list) were for each element we have the pointer to the next element, the label for the table (e.g. TRACKS:PAR) and a pointer to a function which returns the list of columns for that table. This should allow us to avoid any hardcoding.

jgrosseo replied:
@ktf Shall we then keep only the Track lines, the rest should be handled by the default. And if we find a `:` in the name but it is not handled explicitly, this could throw a fatal error

jgrosseo replied:
@pbuehler Could you do this modification, please? Many thanks! Then we merge it.

knopers8 committed:
Improvements in Mergers benchmarks (#4243)

* Fix randomness in FullVsDiff comparison of THnSparse

* Merger benchmark - merge the same number of data regardless of collection size

* use manual time in merger benchmarks

pausing and resuming with the interface methods has too much overhead

* Force memory allocation before benchmarks start

* Comparison of merging, size and (de)serialization of different types

fix csv output

fix parameters for trees

new settings

use entries amount in standard histograms

fix THnSparse segfaults with multiple repetitions

better params for sparse

test boost histograms

a few anti-optimization tricks

use binary_archive for boost serialization - it is faster

cleanups

davidrohr committed:
GPU: Fix compilation when ROOT was compiled with CUDA support

dsekihat committed:
Analysis/PWGDQ: add tableMaker_pp (#4336)

* add tableMaker_pp
* update for tableMaker_pp.cxx
* remove chanrge from DCA

saganatt committed:
DPL Analysis: First version of histogram registry writing (#4343)

ktf committed:
Add ability to code-format a PR. (#4347)

ktf committed:
Push new branch to the alibuild repository

ktf committed:
Update code-formatting.yml

jgrosseo committed:
fixing dynamic calculation of number of clusters (#4346)

* fixing dynamic calculation of number of clusters

* fixing default values

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

mario-krueger committed:
fix small logic bug in track selection (#4350)

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

shahor02 committed:
Register contributions from all sources in vertex-track references

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

ktf committed:
Update code-formatting.yml

aferrero2707 committed:
[MCH] allow sending full HBframes in MCH custom raw page reader (#4286)

This commit adds an option to the MCH CRU page reader that allows to buffer the pages belonging to the same orbit and send them in one single DPL message.
The default behavior is to send each CRU page in a separate DPL message.

ddobrigk committed:
Strangeness tables + strangeness producer, consumer tasks (first commit) (#4316)

ktf committed:
DPL: modernize namespaces (#4349)

pbuehler committed:
Merge branch 'dev' of https://github.com/pbuehler/AliceO2 into temporary

pbuehler committed:
remove  unnecessary code

pbuehler commented:
cleanup!
